Source/DOM classes/Unported or Partial DOM/SVGImageElement.m

```
@@ -81,7 +82,14 @@ - (CALayer *) newLayer
 	frame = CGRectApplyAffineTransform(frame, [SVGHelperUtilities transformAbsoluteIncludingViewportForTransformableOrViewportEstablishingElement:self]);
 	newLayer.frame = frame;
 	
-	NSData *imageData = [NSData dataWithContentsOfURL:[NSURL URLWithString:_href]];
+	NSData *imageData;
+	if( [_href hasPrefix:@"data:"] || [_href hasPrefix:@"http:"] )
+		imageData = [NSData dataWithContentsOfURL:[NSURL URLWithString:_href]];
+	else
+	{
+		NSInputStream *stream = [self.rootOfCurrentDocumentFragment.source sourceFromRelativePath:_href].stream;
+		imageData = [NSData dataWithContentsOfStream:stream initialCapacity:NSUIntegerMax error:nil];
```

Please do something with the NSError parameter.

It is sufficient to DDLogError() it - but we mustn't silently drop errors!